### PR TITLE
Introduce slim BOM

### LINKED ISSUE #635: Provide BoM
**Describe the solution you'd like**
Provide a BoM: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#bill-of-materials-bom-poms to avoid that `java-api` version and `token-client` version do not match.

**Describe alternatives you've considered**
Provide automatic PRs for consumers to upgrade to latest version. Not sure whether this is possible from Github to private Guthub.

### PR DESCRIPTION
**Pull request by liga-oz**
closes #635

@nenaraab please check if `distributionManagement` part is required in this BOM

### CODE REVIEW

File: BOM/pom.xml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.sap.cloud.security.xsuaa</groupId>
```
Comment (nenaraab) [inline BOM/pom.xml:7]: pls remove xsuaa from groupId

File: BOM/pom.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            <email>xsuaa-development@sap.com</email>
+            <organization>SAP SE</organization>
+            <organizationUrl>https://www.sap.com</organizationUrl>
+        </developer>
+    </developers>
+
+    <scm>
+        <connection>scm:git:git://github.com/SAP/cloud-security-xsuaa-integration.git</connection>
+        <developerConnection>sscm:git:git://github.com/SAP/cloud-security-xsuaa-integration.git</developerConnection>
+        <url>http://github.com/SAP/cloud-security-xsuaa-integration/tree/master</url>
```
Comment (nenaraab) [inline BOM/pom.xml:35]: change to main

File: BOM/pom.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    </scm>
+
+    <distributionManagement>
```
Comment (nenaraab) [inline BOM/pom.xml:38]: this part can be removed

File: BOM/pom.xml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            <artifactId>java-security-test</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${parent-groupId}</groupId>
+            <artifactId>java-security</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.sap.cloud.security.xsuaa</groupId>
```
Comment (nenaraab) [inline BOM/pom.xml:98]: no group id specified

File: BOM/pom.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            <artifactId>spring-xsuaa</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>spring-xsuaa-test</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${parent-groupId}</groupId>
```
Comment (nenaraab) [inline BOM/pom.xml:83]: pls double check all group Ids java-security, spring-security and some others have no xsuaa suffix....

**Review by nenaraab — CHANGES_REQUESTED** [review]

File: bom/pom.xml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                <artifactId>spring-xsuaa-test</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>${xsuaa-groupId}</groupId>
+                <artifactId>spring-xsuaa-mock</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>${xsuaa-groupId}</groupId>
```
Comment (nenaraab) [inline bom/pom.xml:96]: This is the only finding I have: please remove the both `spring-xsuaa-it` and `java-security-it` dependencies from here... 👍

**Review by nenaraab — CHANGES_REQUESTED** [review]

File: bom/pom.xml (hunk 6 above)
Comment (nenaraab) [reply bom/pom.xml]: done

**Review by nenaraab — APPROVED** [review]

**Comment by BraunMatthias** [comment]
Great! This will help a lot to avoid version conflicts.

**Pull request merged**